website/server/controllers/api-v3/tasks.js

```diff
@@ -457,7 +458,13 @@ api.updateTask = {
     }
 
     if (sanitizedObj.type === 'daily') {
-      task.isDue = common.shouldDo(Date.now(), sanitizedObj, user.preferences);
+      let optionsForShouldDo = cloneDeep(user.preferences.toObject());
```

Comment:
toObject already clones objects so calling deepClone is not necessary, see https://github.com/Automattic/mongoose/issues/5008